Yoast/wordpress-seo
# Configure imagemin-svgo to handle the SVG accessible attributes.

## Summary

This PR can be summarized in the following changelog entry:

* Configure `imagemin-svgo` to handle the SVG accessible attributes.

With this change, the accessible attributes for the SVG icons in the `images/` directory won't be removed any longer during the build process.

## Relevant technical choices:
- re-enables `imagemin-svgo` after https://github.com/Yoast/wordpress-seo/commit/7005907934677881fce202d94898c9da55ac43f8
- updates `imagemin-svgo` to latest version: we need some new options available in recent svgo versions. As a consequence, other packages get updated see `yarn.lock` /Cc @Xyfi 
- configures the svgo plugins:
  - always removes the `<title>` and `<description>` elements
  - keep ARIA roles and attributes if already set 
  - automatically adds `role="img" aria-hidden="true" focusable="false"` 🎉 by using `addAttributesToSVGElement`, see examples on https://github.com/svg/svgo/blob/d6e462b679a11376694c948c81e5c984c3db8bae/plugins/addAttributesToSVGElement.js
  - svgo seems smart enough to not add doubled roles and attributes 
  - keeps the other config defaults unchanged

Note: `waiting.gif` gets changed as well, but there are no visible changes.

## Test instructions
- check the SVG icons look the same e.g. these ones in the Posts table:
<img width="231" alt="Screenshot 2019-05-01 at 18 55 40" src="https://user-images.githubusercontent.com/1682452/57029727-de04ab00-6c42-11e9-8bef-8b5bc0f1bf5f.png">

- you can also manually view all the SVG icons in `images` dragging them into a tab of your browser or directly in macOS Finder
- unzip and place the attached SVG icon `test.svg` in the `images` directory
- open `test.svg` in your editor of choice and observe the (intentional) mess in the markup
- run `grunt build:images`
- see the new `test.svg` markup is now very clean and has the accessible attributes

## UI changes

Fixes #12654 

[test.svg.zip](https://github.com/Yoast/wordpress-seo/files/3135268/test.svg.zip)
